lib/crud.js

```diff
@@ -79,14 +91,16 @@ function listVersions(entityId, cb) {
   });
 }
 
-function upsert(id, typeId, json, correlationId, done) {
-  if (typeof correlationId === "function") {
-    done = correlationId;
-    correlationId = null;
-  }
-  const entityId = id || uuid.v4();
+function upsert(entity, done) {
+  const entityId = entity.id || uuid.v4();
+  const typeId = entity.type || null;
+  const json = entity.attributes || null;
+  const correlationId = entity.meta.correlationId || null;
+  const relationships = entity.relationships || null;
+
   let newVersionId;
   async.waterfall([
+    (cb) => insertRelations(entityId, relationships, cb),
```

Comment:
Ordering should probably be reversed, i.e. save the entity before the relationships, to avoid having zombie relationships if the second cb fails